Filter out appropriate check_ids from report based on config.checks

kartikp10 opened the pull request:
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

This PR addresses a bug where running checkov like so `checkov -d path/to/iac -c <any-check>` also displays secret scanning related checks. 

This was also reported in #1363 

This also addresses issues with printing code blocks correctly for secret scanning.

Before this change code blocks for secret scanning were blank.

```
Check: CKV_SECRET_19: "Hex High Entropy String"
	FAILED for resource: fc24931058dd40abe8e4719cc5d9be508c7e2585
	File: /terraform/azure/sql.tf:15-16
	Guide: https://docs.bridgecrew.io/docs/git_secrets_19

		15 |
```

After this change:

```
Check: CKV_SECRET_19: "Hex High Entropy String"
	FAILED for resource: fc24931058dd40abe8e4719cc5d9be508c7e2585
	File: /terraform/azure/sql.tf:15-16
	Guide: https://docs.bridgecrew.io/docs/git_secrets_19

		15 |   administrator_login_password = "Aa12345678"
```

gruebel commented:
@kartikp10 just a quick note, this doesn't address issue #1363, because it is related with suppressing a check via a comment.

kartikp10 commented:
> @kartikp10 just a quick note, this doesn't address issue #1363, because it is related with suppressing a check via a comment.

Good catch. I made changes so that it should fix that as well now. This also addresses issues with printing code blocks correctly for secret scanning. 

Before this change code blocks for secret scanning were blank.
```

Check: CKV_SECRET_19: "Hex High Entropy String"
	FAILED for resource: fc24931058dd40abe8e4719cc5d9be508c7e2585
	File: /terraform/azure/sql.tf:15-16
	Guide: https://docs.bridgecrew.io/docs/git_secrets_19

		15 |
```

After this change:

```
Check: CKV_SECRET_19: "Hex High Entropy String"
	FAILED for resource: fc24931058dd40abe8e4719cc5d9be508c7e2585
	File: /terraform/azure/sql.tf:15-16
	Guide: https://docs.bridgecrew.io/docs/git_secrets_19

		15 |   administrator_login_password = "Aa12345678"

```
